engine/function/src/templates/Numeric.ftl

```diff
@@ -1226,13 +1267,38 @@ public class Numeric {
         int n = values.intSize("median");
 
         if (n == 0) {
-            return Double.NaN;
+            return NULL_DOUBLE;
         } else {
-            ${pt.primitive}[] copy = values.copyToArray();
-            Arrays.sort(copy);
-            if (n % 2 == 0)
-                return 0.5 * (copy[n / 2 - 1] + copy[n / 2]);
-            else return copy[n / 2];
+            // NULL values sorted to beginning of the array.
+            ${pt.primitive}[] copy = sort(values.toArray());
+
+            // Determine if there are any NULL in the array.
```

Comment:
I think:
1. We should sort a copy using `Arrays.sort`.
2. We can iterate the array looking for a range of nulls, short-circuiting when we find the first value that is greater than the `null` constant for the type, and then move that range to the beginning if it isn't already at the beginning.
3. We can figure out where NaNs sort to (maybe the end? I hope the end), and check for any to poison with.
4. We can then compute the median if there are no NaNs and non-zero non-null values starting from the end of the non-nulls.